main.tf

```diff
@@ -21,7 +21,7 @@ data "aws_partition" "current" {}
 resource "random_password" "master_password" {
   count = var.create_cluster && var.create_random_password ? 1 : 0
 
-  length  = 10
+  length  = 20
```

Comment:
lets just turn this into a variable with the length that defaults to 10. something like `var.password_length`

Reply:
Why give people a footgun too shoot themselves with? Do you want to allow people to set shorter passwords than 10 chars? Fewer variables == a better API.

There's no cost to having a long password. A 20 char password is good enough for everybody.

Reply:
1. its not for us to dictate to users what their password length should be
2. this change as its written will be a breaking change, essentially rotating the DB master password for all users of the module

I agree in principle that a longer password is a better posture, but we try to not break things for users nor dictate how they should setup their resources. we try to balance between providing a sane set of defaults and good documentation/examples to help them get up and running quickly, but also allowing them flexibility to customize as it suites their setup

Reply:
1. We're currently dictating a password length of 10.
2. The breaking change is a big consideration. I think fixing this insecure default justifies bumping the major version to 6.0.0. If people aren't using version constraints, then they really can't complain about a change like this.

Reply:
@bryantbiggs I think the master password is only used during the creation of the DB and not verified during updates, so it is safe to change it and it won't break anything for existing users except recreation of the `random_password` resource itself. Right?

Reply:
@antonbabenko I am fairly certain it will force the update. The length change will trigger the resource update for the `random_password` which means the input for the password of the cluster will be different and show up as a diff and will get updated on the next apply. should be easy to confirm with the examples

@ryboe this is what we have for the RDS module, it should be a variable https://github.com/terraform-aws-modules/terraform-aws-rds/blob/66336d9c489c4e318799219a7ec5584efdae70be/main.tf#L15

Reply:
It's your call, but I still think it should not be a variable. APIs with no knobs are generally better than APIs with lots of knobs. If the `rds` module exposes it, then it's making a mistake too. The only thing `var.password_length` gives the user is the opportunity to misconfigure the db.

Reply:
What is the status of this PR, is it abandoned? At the moment I "workaround" this by creating my own random_password resource of 32 symbol length and `password = random_password.aurora_master_password.result`